content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -548,7 +548,8 @@ extra:
 
 In order to prevent header spoofing, the authenticating proxy is required to present a valid client
 certificate to the API server for validation against the specified CA before the request headers are
-checked.
+checked. WARNING: make sure to use a distinct CA and do **not** reuse a CA that is used in a different
```

Comment:
I don't think it's not absolutely necessary to use a different CA, but it is important to be careful about these certificates, no matter how they chain.

I would propose that this warning gets turned into a more general "this client certificate will be able to convey authentication info to the API server, so be careful with it."

And then down in the "`--requestheader-allowed-names`" description, we can call out that "any Common Name is allowed, which means that any client certificate issued by any CA in the bundle will be allowed."

Reply:
As far as I understand, the only way it can be secure with the same CA used for both `--client-ca-file` and `--requestheader-client-ca-file` is when absolutely all client traffic goes through the authenticating proxy (i.e. the apiserver is not reachable from other endpoints than the proxy and this `--client-ca-file` "is be used"; I would guess that's a rare setup) **or** when `--requestheader-allowed-names` is chosen very carefully in combination with a procedure that make sure that certificate signing requests with the allowed names are denied in general (i.e. do not `kubectl certificate approve ...` with a CN that matches one of those).

> this client certificate will be able to convey authentication info to the API server, so be careful with it.

I'm worried that this is not clear. Expert users can always choose to make an educated decision against the general recommendation, but to me it seems the only "safe" general recommendation is to use a different CA.

"convey authentication info" is unclear about the actual impact (any client with a valid certificate can effectively escalate privileges to cluster admin). And "be careful" not actionable. What is the recommended, secure default? Where and how do I need to be careful? etc.

Reply:
Hmmm. Yeah, I see your point. Maybe we can go with the "don't do this" language, and just add a "unless you understand the risks and the mechanisms to protect the CA's usage."

WDYT?